pfhedge/features/_base.py

```diff
@@ -85,6 +87,4 @@ class StateIndependentFeature(Feature):
     hedger: None
 
     def of(self: T, derivative: Derivative, hedger: Optional[Module] = None) -> T:
```

Comment:
https://github.com/pfnet-research/pfhedge/pull/343#issuecomment-917744514
I think is the reason is: `super()` implicitly means `super(self, StateIndependentFeature)` where self must be an instance of StateIndependentFeature (https://docs.python.org/ja/3/library/functions.html#super). Since T in `self: T` is bounded to "Feature", mypy gets confused.

The following suggestion should resolve this.

```suggestion
    def of(
        self: "StateIndependentFeature",
        derivative: Derivative,
        hedger: Optional[Module] = None
    ) -> "StateIndependentFeature":
```